lib/ansible/modules/cloud/amazon/ec2_asg.py

```diff
@@ -396,10 +396,24 @@
 
 backoff_params = dict(tries=10, delay=3, backoff=1.5)
 
+
+def paginated_describe_asg(connection, group_name):
+    pg = connection.get_paginator('describe_auto_scaling_groups')
+    for page in pg.paginate(AutoScalingGroupNames=[group_name]):
```

Comment:
is there a reason why you don't do: 

```
return pg.paginate(AutoScalingGroupNames=[group_name]).build_full_result()
```

(other than it being incredibly poorly documented, which is a fair reason)